src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveLoadNonFaultingUnOpTest.template

```diff
@@ -0,0 +1,413 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/******************************************************************************
+ * This file is auto-generated from a template file by the GenerateTests.csx  *
+ * script in tests\src\JIT\HardwareIntrinsics\Arm\Shared. In order to make    *
+ * changes, please update the corresponding template and run according to the *
+ * directions listed in the file.                                             *
+ ******************************************************************************/
+
+using System;
+using System.Numerics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+namespace JIT.HardwareIntrinsics.Arm
+{
+    public static partial class Program
+    {
+        [Fact]
+        public static void {TestName}()
+        {
+            var test = new LoadUnaryOpTest__{TestName}();
+
+            if (test.IsSupported)
+            {
+                // Validates basic functionality works
+                test.RunBasicScenario_Load();
+
+                // Validates basic functionality 
+                test.RunBasicScenario_LoadNonFaulting();
+
+                // Validates calling via reflection works
+                test.RunReflectionScenario_Load();
+
+                // Validates passing an instance member of a class works
+                test.RunClassFldScenario();
+
+                // Validates passing the field of a local struct works
+                test.RunStructLclFldScenario();
+
+                // Validates passing an instance member of a struct works
+                test.RunStructFldScenario();
+
+                // Validates using inside ConditionalSelect with value falseValue
+                // Currently, using this operation in ConditionalSelect() gives incorrect result
+                // when falseReg == targetReg because this instruction uses Pg/Z to update the targetReg
+                // instead of Pg/M to merge it. As such, the value of falseReg is lost. Ideally, such
+                // instructions should be marked similar to RMW (a different flag name) to make sure that
+                // we do not assign falseReg/targetReg same. Then, we would do something like this:
+                //
+                // ldnf1sh target, pg/z, [x0]
+                // sel mask, target, target, falseReg
+                //
+                // This needs more careful thinking, so disabling it for now.
+                // test.ConditionalSelect_FalseOp();
+
+                // Validates using inside ConditionalSelect with zero falseValue
+                test.ConditionalSelect_ZeroOp();
+            }
```

Comment:
Maybe not for this PR, but I remember discussion about testing with addresses that fault. Use an address that is invalid/partially invalid, then test an exception is not raised.
Then maybe add the same test to the normal loads and check it does fault.

Reply:
Yes, @TIHan added that in https://github.com/dotnet/runtime/pull/102180

https://github.com/dotnet/runtime/blob/3750ac51619efbbc59bf07d3879758a9c18c4b0e/src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveLoad2xVectorAndUnzipTest.template#L190-L207